compiler/ilgen/MethodBuilder.cpp

```diff
@@ -522,14 +522,16 @@ MethodBuilder::DefineFunction(const char* const name,
       }   
    MB_REPLAY(");");
 
-   TR::ResolvedMethod *method = new (PERSISTENT_NEW) TR::ResolvedMethod((char*)fileName,
-                                                                        (char*)lineNumber,
-                                                                        (char*)name,
-                                                                        numParms,
-                                                                        parmTypes,
-                                                                        returnType,
-                                                                        entryPoint,
-                                                                        0);
+   // Store ResolvedMethod in the heap region of the TypeDictionary memory segment
+   TR_Memory *memory = typeDictionary()->trMemory();
```

Comment:
Maybe it would be better for the MethodBuilder object to have its own cached _trMemory field, initialized from the TypeDictionary? that way all the allocations could be against the MethodBuilder's memory region and we would then be free to have the MethodBuilder allocate its own memory region in future without having to update all the allocations?

I thought we had discussed this point before, so maybe I'm just not remembering the reason why that's impractical...